x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -276,6 +294,14 @@ private boolean isConcreteRestrictedIndex(String indexPattern) {
         return RestrictedIndicesNames.isRestricted(indexPattern);
     }
 
+    private static boolean authorizeMappingUpdateBwcSpecialCase(Group group, String action) {
+        return action.equals(PutMappingAction.NAME) ||
+                (group.privilege().name().containsAll(IndexPrivilege.CREATE_DOC.name()) ||
+                        group.privilege().name().containsAll(IndexPrivilege.CREATE.name()) ||
+                        group.privilege().name().containsAll(IndexPrivilege.INDEX.name()) ||
+                        group.privilege().name().containsAll(IndexPrivilege.WRITE.name()));
```

Comment:
We'll also need to think about how to write a deprecation warning if this BWC path gets used, but it's complex because we only want to do that if there _isn't_ another `Group` that grants explicit mapping update access.

Reply:
Do we need add version check here? Or is it because this BWC must exist for v8.0 and we cannot check v9.0 yet? If so, can we add `TODO` so it's easier to identify?

Reply:
The bwc logic proved tricky and I ended up restructuring this whole logic.